src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -420,9 +420,7 @@ class Holder {
     }
 
     private String constantValue(Class<?> type, Object value) {
-        if (value instanceof String) {
-            return STR."\{runtimeHelperName()}.LIBRARY_ARENA.allocateFrom(\"\{Utils.quote(Objects.toString(value))}\")";
-        } else if (type == MemorySegment.class) {
+        if (type == MemorySegment.class) {
             return STR."MemorySegment.ofAddress(\{((Number)value).longValue()}L)";
         } else {
```

Comment:
Can we beef up this method with something like `if (value instanceof Number) <current logic> else throw UOE` ?